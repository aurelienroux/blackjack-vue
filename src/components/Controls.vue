<template>
  <div class="controls">
    <div>
      <button class="button__red" @click="getNewDeck">
        New deck
      </button>
      <span test-id="remaining" class="small"> Remaining cards in deck: {{ remainingCards }} </span>
    </div>
    <div>
      <button class="button__red" @click="newHand">
        New hand
      </button>
      <button @click="hitMe('player')" :disabled="!gamePlaying">
        Hit me!
      </button>
      <button @click="dealerPlay" :disabled="!gamePlaying">
        Stay
      </button>
    </div>
  </div>
</template>

<script>
export default {
  name: 'Controls',
  props: ['gamePlaying', 'remainingCards'],
  methods: {
    getNewDeck() {
      this.$emit('getNewDeck')
    },
    newHand() {
      this.$emit('newHand')
    },
    hitMe(player) {
      this.$emit('hitMe', player)
    },
    dealerPlay() {
      this.$emit('dealerPlay')
    }
  }
}
</script>

<style lang="scss">
.controls {
  padding: 0.5rem 2rem;
}

button {
  background: $blue-button;
  border-radius: 5px;
  border: none;
  color: $white;
  cursor: pointer;
  font-family: $font-primary;
  font-size: 0.8rem;
  margin: 0.5rem;
  min-width: 5rem;
  padding: 0.75rem;

  &:disabled {
    background: lighten($blue-button, 15%);
    cursor: not-allowed;
  }

  &.button__red {
    background: $red-button;
  }
}

.small {
  font-size: 0.8rem;
  margin-left: 1rem;
}
</style>
