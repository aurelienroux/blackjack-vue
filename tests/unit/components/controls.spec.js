import { shallowMount } from '@vue/test-utils'
import Controls from '@/components/Controls.vue'

function createLocalWrapper(gamePlaying) {
  return shallowMount(Controls, {
    propsData: {
      gamePlaying: gamePlaying,
      remainingCards: 100
    }
  })
}

describe('Controls component', () => {
  describe('with game on', () => {
    const wrapper = createLocalWrapper(true)
    it('renders', () => {
      expect(wrapper).toMatchSnapshot()
      expect(wrapper.findAll('button').length).toBe(4)
      expect(wrapper.find('[test-id="remaining"]').text()).toMatch('Remaining cards in deck: 100')
    })
  })

  describe('with game off', () => {
    const wrapper = createLocalWrapper(false)
    it('renders', () => {
      expect(wrapper).toMatchSnapshot()
      expect(
        wrapper
          .findAll('button')
          .at(2)
          .attributes('disabled')
      ).toBe('disabled')
      expect(
        wrapper
          .findAll('button')
          .at(3)
          .attributes('disabled')
      ).toBe('disabled')
    })
  })
})
