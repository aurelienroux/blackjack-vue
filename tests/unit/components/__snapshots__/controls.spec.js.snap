// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls component game on renders 1`] = `
<div class="controls">
  <div><button class="button__red">
      New deck
    </button> <span test-id="remaining" class="small"> Remaining cards in deck: 100 </span></div>
  <div><button class="button__red">
      New hand
    </button> <button>
      Hit me
    </button> <button>
      Stay
    </button></div>
</div>
`;

exports[`Controls component game over renders 1`] = `
<div class="controls">
  <div><button class="button__red">
      New deck
    </button> <span test-id="remaining" class="small"> Remaining cards in deck: 100 </span></div>
  <div><button class="button__red">
      New hand
    </button> <button disabled="disabled">
      Hit me
    </button> <button disabled="disabled">
      Stay
    </button></div>
</div>
`;
